packages/vscode-extension/src/debug/depsChecker/vscodeChecker.ts

```diff
@@ -71,22 +77,24 @@ export class VSCodeDepsChecker {
     return await this.depsManager.ensureDependencies(deps, options);
   }
 
-  private async handleLinux(depsStatus: DependencyStatus[]): Promise<boolean> {
+  private async handleLinux(
+    depsStatus: DependencyStatus[]
+  ): Promise<Result<Void, DepsCheckerError>> {
     if (!VSCodeDepsChecker.isLinux()) {
-      return true;
+      return ok(Void);
     }
     const manuallyInstallDeps = depsStatus
       .filter((dep) => !dep.isInstalled)
       .filter((dep) => !dep.details.isLinuxSupported);
 
     if (manuallyInstallDeps.length == 0) {
-      return true;
+      return ok(Void);
     }
 
     const displayMessage = await this.generateLinuxMsg(manuallyInstallDeps);
     await this.display(displayMessage, defaultHelpLink);
     this.logger.cleanup();
-    return false;
+    return err(new DepsCheckerError(displayMessage, defaultHelpLink));
```

Comment:
Consider using `LinuxNotSupportedError` so that have more details to display in the future.

Reply:
Remove the changes in vscode checker and use the LinuxNotSupportedError returned from funcToolChecker / dotnetChecker instead.